src/vulkan-renderer/frame_graph.cpp

```diff
@@ -0,0 +1,498 @@
+
+    std::vector<VkVertexInputAttributeDescription> attribute_bindings;
+    std::vector<VkVertexInputBindingDescription> vertex_bindings;
+    for (const auto *resource : stage->m_reads) {
+        const auto *buffer_resource = dynamic_cast<const BufferResource *>(resource);
+        if (buffer_resource == nullptr) {
+            continue;
+        }
+
+        std::uint32_t binding = stage->m_buffer_bindings.at(buffer_resource);
+        for (auto attribute_binding : buffer_resource->m_vertex_attributes) {
+            attribute_binding.binding = binding;
+            attribute_bindings.push_back(attribute_binding);
+        }
+
+        VkVertexInputBindingDescription vertex_binding = {};
+        vertex_binding.binding = binding;
+        vertex_binding.inputRate = VK_VERTEX_INPUT_RATE_VERTEX;
+        vertex_binding.stride = buffer_resource->m_element_size;
+        vertex_bindings.push_back(vertex_binding);
+    }
+
+    auto vertex_input = wrapper::make_info<VkPipelineVertexInputStateCreateInfo>();
+    vertex_input.vertexAttributeDescriptionCount = static_cast<std::uint32_t>(attribute_bindings.size());
+    vertex_input.vertexBindingDescriptionCount = static_cast<std::uint32_t>(vertex_bindings.size());
+    vertex_input.pVertexAttributeDescriptions = attribute_bindings.data();
+    vertex_input.pVertexBindingDescriptions = vertex_bindings.data();
+
+    // TODO(): Support primitives other than triangles
+    auto input_assembly = wrapper::make_info<VkPipelineInputAssemblyStateCreateInfo>();
+    input_assembly.primitiveRestartEnable = VK_FALSE;
+    input_assembly.topology = VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST;
+
+    auto depth_stencil = wrapper::make_info<VkPipelineDepthStencilStateCreateInfo>();
+    depth_stencil.depthCompareOp = VK_COMPARE_OP_LESS_OR_EQUAL;
+    depth_stencil.depthTestEnable = VK_TRUE;
+    depth_stencil.depthWriteEnable = VK_TRUE;
+
+    // TODO(): Wireframe rendering
+    auto rasterization_state = wrapper::make_info<VkPipelineRasterizationStateCreateInfo>();
+    rasterization_state.cullMode = VK_CULL_MODE_BACK_BIT;
+    rasterization_state.frontFace = VK_FRONT_FACE_CLOCKWISE;
+    rasterization_state.lineWidth = 1.0F;
+    rasterization_state.polygonMode = VK_POLYGON_MODE_FILL;
+
+    // TODO(): Support multisampling again
+    auto multisample_state = wrapper::make_info<VkPipelineMultisampleStateCreateInfo>();
+    multisample_state.minSampleShading = 1.0F;
+    multisample_state.rasterizationSamples = VK_SAMPLE_COUNT_1_BIT;
+
+    VkPipelineColorBlendAttachmentState blend_attachment = {};
+    blend_attachment.colorWriteMask =
+        VK_COLOR_COMPONENT_R_BIT | VK_COLOR_COMPONENT_G_BIT | VK_COLOR_COMPONENT_B_BIT | VK_COLOR_COMPONENT_A_BIT;
+
+    auto blend_state = wrapper::make_info<VkPipelineColorBlendStateCreateInfo>();
+    blend_state.attachmentCount = 1;
+    blend_state.pAttachments = &blend_attachment;
+
+    VkRect2D scissor = {};
```

Comment:
I wouldn't use `= {}`, but just `{}`, as `= {}` may be interpreted than a list initializer. https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es23-prefer-the--initializer-syntax